src/shims/unix/linux/fd.rs

```diff
@@ -208,6 +210,39 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         Ok(Scalar::from_i32(fd))
     }
 
+    /// Currently this function creates new `Socket` without specifying the domain, type, or
+    /// protocol.
+    ///
+    /// This function creates an unnamed socket in the specified domain, of the specified type, and
+    /// using the optionally specified protocol.
+    ///
+    /// The `domain` argument specified a communication domain; this selects the protocol family
+    /// used for communication. The socket `type` specifies the communication semantics. The
+    /// `protocol` specifies a particular protocol to use with the socket. Normally there's only a
+    /// single protocol supported for a particular socket type within a given protocol family, in
+    /// which case `protocol` can be specified as 0. It is possible that many protocols exist and
+    /// in that case, a particular protocol must be specified.
+    ///
+    /// For more information on the arguments see the socket manpage:
+    /// <https://linux.die.net/man/2/socket>
+    fn socket(
+        &mut self,
+        domain: &OpTy<'tcx, Provenance>,
+        type_: &OpTy<'tcx, Provenance>,
+        protocol: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, Scalar<Provenance>> {
+        let this = self.eval_context_mut();
+
+        let _domain = this.read_scalar(domain)?.to_i32()?;
+        let _type_ = this.read_scalar(type_)?.to_i32()?;
+        let _protocol = this.read_scalar(protocol)?.to_i32()?;
```

Comment:
Unless we intend to support *all* values here, we should `throw_unsupported!` for socket types we do not support.